ninja.lua

```diff
@@ -495,6 +516,14 @@ local function compile_file_build(cfg, filecfg, toolset, pch_dependency, regular
 			cxxflags = {"CXXFLAGS = $CXXFLAGS " .. getcxxflags(toolset, cfg, filecfg)}
 		end
 		add_build(cfg, objfilename, {}, "cxx", {filepath}, pch_dependency, regular_file_dependencies, cxxflags)
+	elseif shouldcompileasasm(filecfg) and toolset == p.tools.msc then
```

Comment:
Why `toolset` filtering here?
I would prefer to handle "asm" in `p.tools.gcc.gettoolname` (to return "`gcc`").